java/src/jmri/jmrit/operations/rollingstock/cars/CarManager.java

```diff
@@ -309,20 +309,33 @@ public List<Car> getByTrainDestinationList(Train train) {
                 out.add(car); // place at end of list
                 lastCarsIndex++;
             } else if (car.isPassenger()) {
-                // block passenger cars at end of list, but before cabooses or car with FRED
-                int index;
-                for (index = 0; index < lastCarsIndex; index++) {
-                    Car carTest = out.get(out.size() - 1 - index);
-                    log.debug("Car ({}) has blocking number: {}", carTest.toString(), carTest.getBlocking());
-                    if (carTest.isPassenger() &&
-                            !carTest.isCaboose() &&
-                            !carTest.hasFred() &&
-                            carTest.getBlocking() < car.getBlocking()) {
-                        break;
+                if (car.getBlocking() < 0) {
+                 // block passenger cars with negative blocking numbers at front of train
```

Comment:
No need to fix, but indentation is wrong for this comment, making the code a little bit harder to read.